backend/src/user/auth.service.ts

```diff
@@ -1,22 +1,28 @@
 import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
 import { JwtService } from '@nestjs/jwt';
 
 import { Request, Response } from 'express';
-import { addUser, getUser } from '../db/mock/usersMock';
+import { User, UserDocument } from 'src/db/schemas/user.schema';
+import { CreateUserDto } from 'src/db/dto/create-user.dto';
 
 @Injectable()
 export class AuthService {
-  constructor(private readonly jwtService: JwtService) {}
+  constructor(private readonly jwtService: JwtService,
+  @InjectModel(User.name) private readonly model: Model<UserDocument>,
+  ) {}
 
-  generateAuth(req: Request, res: Response): void {
+  async generateAuth(req: Request, res: Response): Promise<void> {
     if (req.body && req.body.username && req.body.password) {
       const { username, password } = req.body;
-      const userObj = getUser({ username, password });
-
-      if (userObj) {
-        const accessToken = this.jwtService.sign(userObj, {
+      const userObj = await this.getUser(username);
+      console.log('User fetched!', userObj);
+      
+      if (userObj && userObj.password === password) {
+        const accessToken = this.jwtService.sign({username:userObj.username}, {
           // algorithm: 'RS256',
-          expiresIn: 3 * 24 * 3600,
+          expiresIn: 3600,
```

Comment:
Please remove this, this was a requested change from frontend, as it's not very UX-friendly for user to keep getting kicked out every hour.

Reply:
@Substancia let's make it `3*24*3600` itself ?

Reply:
Yes please.